packages/legend-graph/src/graph-manager/protocol/pure/v1/transformation/pureGraph/to/helpers/V1_ValueSpecificationBuilderHelper.ts

```diff
@@ -766,11 +795,40 @@ function buildPropertyGraphFetchTree(
     processingContext.flushVariable('this');
     processingContext.pop();
   } else {
-    property = V1_getAppliedProperty(
-      parentClass,
-      undefined,
-      propertyGraphFetchTree.property,
-    );
+    //While building property graph fetch tree, the Class of root graph
```

Comment:
Thanks for the clarification here, what I don't understand is, why do we just do this when building execution plan, but not when building the value specification for query